backend/imagekit/imagekit.go

```diff
@@ -0,0 +1,794 @@
+
+	if err != nil {
+		return err
+	}
+
+	o.file = *file
+
+	return nil
+}
+
+// Removes this object
+func (o *Object) Remove(ctx context.Context) error {
+
+	fs.Debugf(o.remote, "Removing file: %s", o.file.FileId)
+
+	_, e := o.fs.ik.DeleteFile(ctx, o.file.FileId)
+
+	if e != nil {
+		return e
+	}
+
+	return nil
+}
+
+// SetModTime sets the metadata on the object to set the modification date
+func (o *Object) SetModTime(ctx context.Context, t time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+func uploadFile(ctx context.Context, f *Fs, in io.Reader, srcRemote string, options ...fs.OpenOption) (fs.Object, error) {
+
+	fs.Debugf(srcRemote, "Uploading file")
+
+	remote := path.Join(f.root, srcRemote)
+	folderPath := f.EncodePath(remote[:strings.LastIndex(remote, "/")+1])
+	fileName := f.EncodeFileName(remote[strings.LastIndex(remote, "/")+1:])
+
+	UseUniqueFileName := new(bool)
+	*UseUniqueFileName = false
+
+	_, _, err := f.ik.Upload(ctx, in, client.UploadParam{
+		FileName:          fileName,
+		Folder:            folderPath,
+		UseUniqueFileName: UseUniqueFileName,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return f.NewObject(ctx, srcRemote)
+}
+
+func (o *Object) readMetaData(ctx context.Context) error {
+
+	if o.meta != nil {
+		return nil
+	}
+
+	meta := make(map[string]string)
+
+	meta["btime"] = o.file.CreatedAt.Format(time.RFC3339)
+
+	meta["size"] = strconv.FormatUint(o.file.Size, 10)
+
+	meta["fileType"] = string(o.file.FileType)
+	meta["height"] = strconv.Itoa(o.file.Height)
+	meta["width"] = strconv.Itoa(o.file.Width)
+	meta["hasAlpha"] = strconv.FormatBool(o.file.HasAlpha)
+
+	for k, v := range o.file.EmbeddedMetadata {
+		switch v := v.(type) {
+		case string:
+			meta[k] = v
+		case float64:
+			meta[k] = strconv.FormatFloat(v, 'f', -1, 64)
+		case bool:
+			meta[k] = strconv.FormatBool(v)
+		default:
+			meta[k] = fmt.Sprintf("%v", v)
+		}
+	}
+
+	if o.file.Tags != nil {
+		meta["tags"] = strings.Join(o.file.Tags, ",")
+	}
+
+	if o.file.CustomCoordinates != nil {
+		meta["customCoordinates"] = *o.file.CustomCoordinates
+	}
+
+	if o.file.IsPrivateFile != nil {
+		meta["isPrivateFile"] = strconv.FormatBool(*o.file.IsPrivateFile)
+	}
+
+	if o.file.AITags != nil {
+		googleTags := []string{}
+		awsTags := []string{}
+
+		for _, tag := range o.file.AITags {
+			if tag.Source == "google-auto-tagging" {
+				googleTags = append(googleTags, tag.Name)
+			} else if tag.Source == "aws-auto-tagging" {
+				awsTags = append(awsTags, tag.Name)
+			}
+		}
+
+		if len(googleTags) > 0 {
+			meta["googleTags"] = strings.Join(googleTags, ",")
+		}
+
+		if len(awsTags) > 0 {
+			meta["awsTags"] = strings.Join(awsTags, ",")
+		}
+	}
+
+	o.meta = meta
+
+	return nil
+
+}
+
+func (o *Object) Metadata(ctx context.Context) (metadata fs.Metadata, err error) {
+
+	err = o.readMetaData(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata = make(fs.Metadata, len(o.meta))
+
+	for k, v := range o.meta {
```

Comment:
why not keep the metadata in the object as `fs.Metadata`? It would save a conversion.